### PR TITLE
ACL parsing from YAML, and validation against JSON schema

### PR DESCRIPTION
**Pull request by mike-north**
Includes minimal test cases to cover "owner ACLs" and "release owner ACLs"

### CODE REVIEW

File: src/models/acl.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  },
+  required: ['paths'],
+};
+
+function parseYaml(
+  yamlText: string,
+): IAclBase & ({ owners?: string[]; release_owners?: string[] }) {
+  const aclData = yml.safeLoad(yamlText);
+  const result = validator.validate(aclData, ACL_DATA_SCHEMA);
+  if (result.valid)
```
Comment (chadhietala) [inline src/models/acl.ts]: Can this be turned into a type predicate so you don't have explicitly cast?

File: src/models/acl.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      type: 'array',
+      items: { type: 'string' },
+    },
+  },
+  required: ['paths'],
+};
+
+function parseYaml(
+  yamlText: string,
+): IAclBase & ({ owners?: string[]; release_owners?: string[] }) {
```
Comment (chadhietala) [inline src/models/acl.ts]: Can you hoist out this union into it's own `type`

File: src/models/acl.ts (hunk 1 above)
Comment (mike-north) [reply src/models/acl.ts]: Addressed in #26

File: src/models/acl.ts (hunk 2 above)
Comment (mike-north) [reply src/models/acl.ts]: Addressed in #26

**Pull request merged**